src/jit_channel/channel_manager.rs

```diff
@@ -216,20 +228,68 @@ impl OutboundJITChannelState {
 		}
 	}
 
-	pub fn htlc_intercepted(
-		&self, expected_outbound_amount_msat: u64, intercept_id: InterceptId,
-	) -> Result<Self, ChannelStateError> {
+	pub fn htlc_intercepted(&self, htlc: InterceptedHTLC) -> Result<Self, ChannelStateError> {
 		match self {
-			OutboundJITChannelState::InvoiceParametersGenerated { opening_fee_params, .. } => {
-				compute_opening_fee(
-					expected_outbound_amount_msat,
-					opening_fee_params.min_fee_msat,
-					opening_fee_params.proportional,
-				).map(|opening_fee_msat| OutboundJITChannelState::PendingChannelOpen {
-					intercept_id,
-					opening_fee_msat,
-					amt_to_forward_msat: expected_outbound_amount_msat - opening_fee_msat,
-				}).ok_or(ChannelStateError(format!("Could not compute valid opening fee with min_fee_msat = {}, proportional = {}, and expected_outbound_amount_msat = {}", opening_fee_params.min_fee_msat, opening_fee_params.proportional, expected_outbound_amount_msat)))
+			OutboundJITChannelState::InvoiceParametersGenerated {
+				opening_fee_params,
+				payment_size_msat,
+				..
+			} => {
+				let htlcs = vec![htlc];
+				let supports_mpp = payment_size_msat.is_some();
+				let payment_size_msat =
+					payment_size_msat.unwrap_or(htlc.expected_outbound_amount_msat);
+				let opening_fee_msat = compute_opening_fee(payment_size_msat, opening_fee_params.min_fee_msat, opening_fee_params.proportional).ok_or(ChannelStateError(format!("Could not compute valid opening fee with min_fee_msat = {}, proportional = {}, and expected_outbound_amount_msat = {}", opening_fee_params.min_fee_msat, opening_fee_params.proportional, htlc.expected_outbound_amount_msat)))?;
+				let amt_to_forward_msat = payment_size_msat - opening_fee_msat;
+
+				if htlc.expected_outbound_amount_msat >= payment_size_msat
+					&& amt_to_forward_msat > 0
+				{
+					Ok(OutboundJITChannelState::PendingChannelOpen {
+						htlcs,
+						opening_fee_msat,
+						amt_to_forward_msat,
+					})
+				} else {
+					if supports_mpp {
+						Ok(OutboundJITChannelState::PendingPaymentReceived {
+							htlcs,
+							amt_to_forward_msat,
+							opening_fee_msat,
+							payment_size_msat,
+						})
+					} else {
+						Err(ChannelStateError("HTLC is too small to pay opening fee".to_string()))
+					}
+				}
+			}
+			OutboundJITChannelState::PendingPaymentReceived {
+				htlcs,
+				payment_size_msat,
+				amt_to_forward_msat,
+				opening_fee_msat,
+			} => {
+				let mut htlcs = htlcs.clone();
+				htlcs.push(htlc);
+
+				let total_expected_outbound_amount_msat = htlcs
+					.iter()
+					.fold(0, |total_msat, htlc| total_msat + htlc.expected_outbound_amount_msat);
```

Comment:
nit: Could write this a tad simpler:
```suggestion
					.map(|htlc| htlc.expected_outbound_amount_msat).sum();
```

Reply:
Easier to read but probably _slightly_ less efficient unless the compiler does some magic to avoid two loops? I'll go with easier to read regardless because the # of htlcs is tiny.